uint8 id
float32 angle
float64 x
float64 y
float64 z